packages/next/src/shared/lib/router/router.ts

```diff
@@ -498,7 +498,8 @@ function fetchNextData({
       headers: Object.assign(
         {} as HeadersInit,
         isPrefetch ? { purpose: 'prefetch' } : {},
-        isPrefetch && hasMiddleware ? { 'x-middleware-prefetch': '1' } : {}
+        isPrefetch && hasMiddleware ? { 'x-middleware-prefetch': '1' } : {},
+        { 'x-deployment-id': process.env.NEXT_DEPLOYMENT_ID }
```

Comment:
```suggestion
        process.env.NEXT_DEPLOYMENT_ID ? { 'x-deployment-id': process.env.NEXT_DEPLOYMENT_ID } : {}
```
We should only set the header if it has a value.